scylla/src/transport/session_test.rs

```diff
@@ -913,3 +913,83 @@ async fn test_await_timed_schema_agreement() {
         .await
         .unwrap();
 }
+
+#[tokio::test]
+async fn test_timestamp() {
+    let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+    let session = SessionBuilder::new().known_node(uri).build().await.unwrap();
+
+    session.query("CREATE KEYSPACE IF NOT EXISTS ks WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor' : 1}", &[]).await.unwrap();
+    session
+        .query("DROP TABLE IF EXISTS ks.t_timestamp;", &[])
+        .await
+        .unwrap();
+    session
+        .query(
+            "CREATE TABLE IF NOT EXISTS ks.t_timestamp (a int, b int, c text, primary key (a, b))",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    session.await_schema_agreement().await.unwrap();
+
+    let query_str = "INSERT INTO ks.t_timestamp (a, b, c) VALUES (?, ?, ?)";
+
+    // test regular query timestamps
+
+    let mut regular_query = Query::new(query_str.to_string());
+
+    regular_query.set_timestamp(Some(420));
+    session
+        .query(regular_query.clone(), (2_i32, 2_i32, "higher timestamp"))
+        .await
+        .unwrap();
+
+    regular_query.set_timestamp(Some(42));
+    session
+        .query(regular_query, (2_i32, 2_i32, "lower timestamp"))
+        .await
+        .unwrap();
+
+    // test prepared statement timestamps
+
+    let mut prepared_statement = session.prepare(query_str).await.unwrap();
+
+    prepared_statement.set_timestamp(Some(420));
+    session
+        .execute(&prepared_statement, (1_i32, 2_i32, "higher timestamp"))
+        .await
+        .unwrap();
+
+    prepared_statement.set_timestamp(Some(42));
+    session
+        .execute(&prepared_statement, (1_i32, 2_i32, "lower timestamp"))
+        .await
+        .unwrap();
+
+    let rs = session
+        .query("SELECT a, b, c FROM ks.t_timestamp", &[])
+        .await
+        .unwrap()
+        .rows
+        .unwrap();
+
+    let mut results: Vec<(i32, i32, &String)> = rs
+        .iter()
+        .map(|r| {
+            let a = r.columns[0].as_ref().unwrap().as_int().unwrap();
+            let b = r.columns[1].as_ref().unwrap().as_int().unwrap();
+            let c = r.columns[2].as_ref().unwrap().as_text().unwrap();
+            (a, b, c)
+        })
```

Comment:
Couldn't `into_typed` be used here instead of casting the types manually?

Reply:
It could, fixed